coffea/jetmet_tools/CorrectedJetsFactory.py

```diff
@@ -220,28 +365,80 @@ def build(self, jets, lazy_cache):
         out_dict[self.name_map["JetMass"] + "_jec"] = out_dict[self.name_map["JetMass"]]
 
         # in jer we need to have a stash for the intermediate JEC products
-        has_jer = False
-        if self.jec_stack.jer is not None and self.jec_stack.jersf is not None:
+        if self.tool == "jecstack":
+            has_jer = False
+            if self.jec_stack.jer is not None and self.jec_stack.jersf is not None:
+                has_jer = True
+        elif self.tool == "clib":
             has_jer = True
+
+        if has_jer:
             jer_name_map = dict(self.name_map)
             jer_name_map["JetPt"] = jer_name_map["JetPt"] + "_jec"
             jer_name_map["JetMass"] = jer_name_map["JetMass"] + "_jec"
 
-            jerargs = {
-                k: out_dict[jer_name_map[k]] for k in self.jec_stack.jer.signature
-            }
-            out_dict["jet_energy_resolution"] = self.jec_stack.jer.getResolution(
-                **jerargs, form=scalar_form, lazy_cache=lazy_cache
-            )
+            if self.tool == "jecstack":
+                jerargs = {
+                    k: out_dict[jer_name_map[k]] for k in self.jec_stack.jer.signature
+                }
+                out_dict["jet_energy_resolution"] = self.jec_stack.jer.getResolution(
+                    **jerargs, form=scalar_form, lazy_cache=lazy_cache
+                )
 
-            jersfargs = {
-                k: out_dict[jer_name_map[k]] for k in self.jec_stack.jersf.signature
-            }
-            out_dict[
-                "jet_energy_resolution_scale_factor"
-            ] = self.jec_stack.jersf.getScaleFactor(
-                **jersfargs, form=_JERSF_FORM, lazy_cache=lazy_cache
-            )
+                jersfargs = {
+                    k: out_dict[jer_name_map[k]] for k in self.jec_stack.jersf.signature
+                }
+                out_dict[
+                    "jet_energy_resolution_scale_factor"
+                ] = self.jec_stack.jersf.getScaleFactor(
+                    **jersfargs, form=_JERSF_FORM, lazy_cache=lazy_cache
+                )
+            elif self.tool == "clib":
+                ## needed to attach to jets the JECs
+                jer_out_parms = out.layout.parameters
+                jer_out_parms["corrected"] = True
```

Comment:
Consider to add a method here:
https://github.com/CoffeaTeam/coffea/blob/master/src/coffea/nanoevents/methods/nanoaod.py#L499